### PR TITLE
Exponential emergence

### PR DESCRIPTION
**Pull request by user624086**
exponential like emergence scrore

### CODE REVIEW

File: scripts/algorithms/emergence.py [hunk 1]
```diff
@@ -157,6 +157,28 @@ def escore2(self, show=False, term = None):
             score = fit_score(normalized_term, y_fit)
         return  trend[0] if abs(trend[0]) >= 0.001 else trend[1]
 
+    def escore_exponential(self, weekly_values):
+        # todo: Modify not to use weekly_values
+        # todo: Create -exp parameter, e.g. power of weight function (currently linear = 1)
+
+        # exponential like emergence score
```
Comment (IanGrimstead) [inline scripts/algorithms/emergence.py]:
```suggestion
```

File: scripts/algorithms/emergence.py [hunk 2]
```diff
@@ -157,6 +157,28 @@ def escore2(self, show=False, term = None):
...
+        my_weekly_values = weekly_values.copy()
+        # convert into whole multiples of 52 (weeks)
+        weeks_in_year = 52
+        num_years = len(my_weekly_values) // weeks_in_year
```
Comment (IanGrimstead) [inline scripts/algorithms/emergence.py]:
```suggestion
        num_complete_years = len(my_weekly_values) // weeks_in_year
```

File: scripts/algorithms/emergence.py [hunk 3]
```diff
@@ -157,6 +157,28 @@ def escore2(self, show=False, term = None):
...
+        my_weekly_values = my_weekly_values[-num_years * weeks_in_year:]
```
Comment (IanGrimstead) [inline scripts/algorithms/emergence.py]:
```suggestion
        my_weekly_values = my_weekly_values[-num_complete_years * weeks_in_year:]
```

File: scripts/algorithms/emergence.py [hunk 4]
```diff
@@ -157,6 +157,28 @@ def escore2(self, show=False, term = None):
...
+        # convert to table with yearly values as rows, then sum rows to obtain yearly values
+        yearly_values = np.array(my_weekly_values).reshape(-1, weeks_in_year)
```
Comment (IanGrimstead) [inline scripts/algorithms/emergence.py]:
```suggestion
        weekly_by_year_values = np.array(my_weekly_values).reshape(-1, weeks_in_year)
```

File: scripts/algorithms/emergence.py [hunk 5]
```diff
@@ -157,6 +157,28 @@ def escore2(self, show=False, term = None):
...
+        yearly_values = [sum(i) for i in yearly_values]
```
Comment (IanGrimstead) [inline scripts/algorithms/emergence.py]:
```suggestion
        yearly_values = [sum(i) for i in weekly_by_year_values]
```

File: scripts/algorithms/emergence.py [hunk 6]
```diff
@@ -157,6 +157,28 @@ def escore2(self, show=False, term = None):
...
+        # calculate score
```
Comment (IanGrimstead) [inline scripts/algorithms/emergence.py]:
```suggestion
```

File: scripts/algorithms/emergence.py [hunk 7]
```diff
@@ -157,6 +157,28 @@ def escore2(self, show=False, term = None):
             score = fit_score(normalized_term, y_fit)
         return  trend[0] if abs(trend[0]) >= 0.001 else trend[1]
 
+    def escore_exponential(self, weekly_values):
```
Comment (IanGrimstead) [inline scripts/algorithms/emergence.py]: Maybe a clearer name - its linear not exponential?

File: scripts/pipeline.py [hunk 8]
```diff
@@ -189,16 +191,22 @@ def pickle_object(short_name, obj):
             if max(quarterly_values) < minimum_patents_per_quarter:
                 continue
 
-            if em.init_vars(row_indices, row_values, porter=not curves):
-                escore = em.calculate_escore() if not curves else em.escore2()
+            if em.init_vars(row_indices, row_values):
+                if exponential:
```
Comment (IanGrimstead) [inline scripts/pipeline.py]: Move `weekly_value` definition here (or move work into `escore_exponential` method)

File: scripts/algorithms/emergence.py (hunk 7 above)
Comment (user624086) [reply scripts/algorithms/emergence.py]: I think its OK, having thought it through again, but am open to correction.

File: scripts/algorithms/emergence.py (hunk 1 above)
Comment (user624086) [reply scripts/algorithms/emergence.py]: I've moved this to the function documentation line to inform that's it's not a real exponential

File: scripts/algorithms/emergence.py (hunk 6 above)
Comment (user624086) [reply scripts/algorithms/emergence.py]: I turned this into a more descriptive comment

**Pull request merged**